json.go

```diff
@@ -190,15 +189,11 @@ func appendJSON(buf []byte, v interface{}) ([]byte, error) {
 		}
 		return appendFloat(buf, t, 64), nil
 	case string:
-		if !utf8.ValidString(t) {
-			// the next line replaces invalid characters.
-			t = strings.ToValidUTF8(t, string(utf8.RuneError))
-		}
 		// escaped length = 2*len(t) + 2 double quotes
 		if cap(buf) < (len(t)*2 + 2) {
```

Comment:
We also need to update the estimation of the buffer usage to `(len(t)*6 + 2)`.

We should not append data to `buf` when `buf` needs to be extended for the extra data.
Instead, we should return `(nil, ErrTooLarge)`.
https://github.com/cybozu-go/log/blob/v1.6.1/formatter.go#L16

`(len(t)*6 + 2)` would be an overestimation in most cases.
Will this change seriously increase the risk of errors in logging?
I suppose "No."
The size of `buf` is 1 MiB and it would be sufficient even for the new estimation.
https://github.com/cybozu-go/log/blob/v1.6.1/constants.go#L45

Reply:
All these capacity checks are broken since implemented, see #32.

Reply:
WTH! Let's fix estimations in another PR.